src/identity/main.cairo

```diff
@@ -88,17 +174,164 @@ mod Identity {
+                    )
+                )
+        }
+
+        fn set_extended_verifier_data(
+            ref self: ContractState, id: u128, field: felt252, data: Span<felt252>, domain: u32
+        ) {
+            let verifier = get_caller_address();
+            self
+                .set(
+                    VERIFIER_DATA_ADDR,
+                    array![id.into(), field, verifier.into()].span(),
+                    0,
+                    data,
+                    domain
+                );
+            self
+                .emit(
+                    Event::ExtendedVerifierDataUpdate(
+                        ExtendedVerifierDataUpdate { id, field, _data: data, verifier, }
+                    )
+                );
+        }
+    }
+
+    //
+    // Internals
+    //
+
+    #[generate_trait]
+    impl InternalImpl of InternalTrait {
+        // todo: move these functions into a storage contract when components are available
+        fn get_extended(
+            self: @ContractState,
+            fn_name: felt252,
+            params: Span<felt252>,
+            offset: u8,
+            length: felt252,
+            domain: u32
+        ) -> Span<felt252> {
+            let base = self.compute_base_address(fn_name, params);
+            let mut data = ArrayTrait::new();
+            let mut offset = 0;
+            loop {
+                if length == offset.into() {
+                    break ();
+                }
+                let value = self._get(domain, base, offset);
+                data.append(value);
+                offset += 1;
+            };
+            data.span()
+        }
+
+        fn get_unbounded(
+            self: @ContractState, fn_name: felt252, params: Span<felt252>, offset: u8, domain: u32
+        ) -> Span<felt252> {
+            let base = self.compute_base_address(fn_name, params);
+            let mut data = ArrayTrait::new();
+            let mut offset = 0;
+            loop {
+                let value = self._get(domain, base, offset);
+                if value == 0 {
+                    break ();
+                }
+                data.append(value);
+                offset += 1;
+            };
+            data.span()
+        }
+
+        fn _get(
+            self: @ContractState, domain: u32, base: starknet::StorageBaseAddress, offset: u8,
+        ) -> felt252 {
+            starknet::storage_read_syscall(
+                domain, starknet::storage_address_from_base_and_offset(base, offset)
+            )
+                .unwrap_syscall()
+        }
+
+        fn set(
+            ref self: ContractState,
+            fn_name: felt252,
+            params: Span<felt252>,
+            offset: u8,
+            value: Span<felt252>,
+            domain: u32,
+        ) {
+            let base = self.compute_base_address(fn_name, params);
+            self._set(domain, base, value, offset: offset);
+        }
+
+        fn _set(
```

Comment:
I would like to get a better answer from Core Stars about that but from my understanding of the source code of Cairo, final addresses are just computed as `base_address` + `offset`. Which means two things:
1) In this code, our array value cannot be bigger than 256 felts (should usually be enough but it would be better to support usize if we can)
2) We can probably optimize it for gas a little bit. offset is redundant with value.

This needs to be tested but I think this code would be equivalent:
```cairo
        fn _set(ref self: ContractState, domain: u32, base: felt252, mut values: Span<felt252>,) {
            match values.pop_back() {
                Option::Some(value) => {
                    let addr = starknet::storage_base_address_from_felt252(
                        base + values.len().into()
                    );
                    starknet::storage_write_syscall(
                        domain, starknet::storage_address_from_base_and_offset(addr, 0), *value
                    );
                    self._set(domain, base, values);
                },
                Option::None => {}
            }
        }
```